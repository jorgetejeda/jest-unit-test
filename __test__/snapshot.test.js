import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe("Es hora de las instantaneas", () => {
  test("Snapshot ", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });
  test("Siempre fallara la instantanea", () => {
    const user = {
      id: Math.floor(Math.random() * 100),
      name: "Jorge Adames",
      profesion: "Javascript Developer",
      createAt: new Date(),
    };
    expect(user).toMatchSnapshot();
  });
  test("Tenemos una excepcion dentro del codigo", () => {
    const user = {
      id: Math.floor(Math.random() * 100),
      name: "Dominique Victoriano",
      createAt: new Date(),
    };
    expect(user).toMatchSnapshot({
      //Como el id es muy cambiante le agregamos una excepcion
      //A esta propiedad en especifico
      id: expect.any(Number),
      createAt: expect.any(Date),
    });
  });
});
