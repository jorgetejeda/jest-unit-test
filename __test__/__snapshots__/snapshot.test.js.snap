// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2021-06-29T02:42:23.546Z,
  "id": 59,
  "name": "Jorge Adames",
  "profesion": "Javascript Developer",
}
`;

exports[`Es hora de las instantaneas Snapshot  1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepcion dentro del codigo 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
  "name": "Dominique Victoriano",
}
`;
